File: delivery-system-server/src/main/java/ru/delivery/system/model/json/cargo/CargoInfoResponse.java
package ru.delivery.system.model.json.cargo;

import lombok.Getter;
import lombok.Setter;
import ru.delivery.system.common.enums.ProductStatus;

import javax.json.bind.annotation.JsonbProperty;
import java.util.Date;
import java.util.List;

public class CargoInfoResponse {

    @JsonbProperty("cargoList")
    @Getter @Setter
    private List<Cargo> cargoList;

    public static class Cargo {

        /**
         * Id самого товара
         */
        @JsonbProperty("cargoId")
        @Getter @Setter
        private Integer cargoId;

        /**
         * Id артикула продукции
         */
        @JsonbProperty("productId")
        @Getter @Setter
        private Integer productId;

        /**
         * Номер склада, на котором хранится товар
         */
        @JsonbProperty("warehouseId")
        @Getter @Setter
        private Integer warehouseId;

        /**
         * Список номеров заказов, в которых учавствовал товар
         */
        @JsonbProperty("orderIdList")
        @Getter @Setter
        private List<Integer> orderIdList;

        /**
         * Стаутс товара
         *
         * @see ProductStatus
         */
        @JsonbProperty("status")
        @Getter @Setter
        private String status;

        /**
         * История изменений статустов конкретного товара
         */
        @JsonbProperty("movementsList")
        @Getter @Setter
        private List<Movement> movementsList;

        public static class Movement {
            @JsonbProperty("movementId")
            @Getter @Setter
            private Integer movementId;

            @JsonbProperty("newStatus")
            @Getter @Setter
            private String newStatus;

            @JsonbProperty("prevStatus")
            @Getter @Setter
            private String prevStatus;

            @JsonbProperty("date")
            @Getter @Setter
            private Date date;
        }
    }
}

File: delivery-system-desktop/src/main/kotlin/ru/delivery/system/models/json/CargoInfoResponse.kt
package ru.delivery.system.models.json

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*
import kotlin.collections.ArrayList

class CargoInfoResponse {

    @JsonProperty("cargoList")
    var cargoList: List<Cargo>? = null


}

@JsonInclude(JsonInclude.Include.NON_NULL)
class Cargo {

    /**
     * Id самого товара
     */
    @JsonProperty("cargoId")
    var cargoId: Int? = null

    /**
     * Id артикула продукции
     */
    @JsonProperty("productId")
    var productId: Int? = null

    /**
     * Номер склада, на котором хранится товар
     */
    @JsonProperty("warehouseId")
    var warehouseId: Int? = null

    /**
     * Список номеров заказов, в которых учавствовал товар
     */
    @JsonProperty("orderIdList")
    var orderIdList: List<Int> = ArrayList()

    /**
     * Стаутс товара
     *
     * @see ProductStatus
     */
    @JsonProperty("status")
    var status: String? = null

    /**
     * История изменений статустов конкретного товара
     */
    @JsonProperty("movementsList")
    var movementsList: List<Movement> = ArrayList()

    @JsonInclude(JsonInclude.Include.NON_NULL)
    class Movement {
        @JsonProperty("movementId")
        var movementId: Int? = null

        @JsonProperty("newStatus")
        var newStatus: String? = null

        @JsonProperty("prevStatus")
        var prevStatus: String? = null

        @JsonProperty("date")
        var date: Date? = null
    }
}
